dds_web/api/user.py

```diff
@@ -874,31 +874,51 @@ def post(self):
         # Check if email is registered to a user
         try:
             existing_user = user_schemas.UserSchema().load({"email": user_email})
+            unanswered_invite = user_schemas.UnansweredInvite().load({"email": user_email})
         except sqlalchemy.exc.OperationalError as err:
             raise ddserr.DatabaseError(message=str(err), alt_message="Unexpected database error.")
 
         if not existing_user:
-            raise ddserr.NoSuchUserError(
-                f"The user with email '{user_email}' does not have access to the specified project."
-                " Cannot remove non-existent project access."
-            )
+            if not unanswered_invite:
+                # The user doesn't exist and doesnt have a pending invite either
+                raise ddserr.NoSuchUserError(
+                    f"The user with email '{user_email}' does not have access to the specified project."
+                    " Cannot remove non-existent project access."
+                )
+            else:
+                invite_id = unanswered_invite.id
+                project_invite_key = models.ProjectInviteKeys.query.filter_by(
+                    invite_id=invite_id, project_id=project.id
+                ).one_or_none()
 
-        user_in_project = False
-        for user_association in project.researchusers:
-            if user_association.user_id == existing_user.username:
-                user_in_project = True
-                db.session.delete(user_association)
-                project_user_key = models.ProjectUserKeys.query.filter_by(
-                    project_id=project.id, user_id=existing_user.username
-                ).first()
-                if project_user_key:
-                    db.session.delete(project_user_key)
-
-        if not user_in_project:
-            raise ddserr.NoSuchUserError(
-                f"The user with email '{user_email}' does not have access to the specified project."
-                " Cannot remove non-existent project access."
-            )
+                # if the unasnwsered invite was asociated in the project, remove the invite and the asociation
+                if project_invite_key:
+                    db.session.delete(unanswered_invite)
+                    db.session.delete(project_invite_key)
```

Comment:
We only want to delete the unanswered invite if it's only associated with one (this) single project. So you should either count the number of project_invite_keys (`unanswered_invite.project_invite_keys`) or first delete the `project_invite_key` and then check if the invite has any more.

Reply:
Oh, I didn't though of that scenario sorry